assets/blocks/course-outline/lesson-block/edit.js

```diff
@@ -123,7 +127,12 @@ export const EditLessonBlock = ( props ) => {
 			<LessonBlockSettings
 				{ ...props }
 				previewStatus={ previewStatus }
-				setPreviewStatus={ setPreviewStatus }
+				setPreviewStatus={ ( status ) => {
+					dispatch( COURSE_STATUS_STORE ).setLessonStatus(
```

Comment:
Hmm it fits in this file since we already use `useDispatch` however I don't think that we should show preference to one of the two in general.

If I understand this correctly `useDispatch` is a wrapper of `dispatch` which means that it doesn't add any extra functionality and simply imposes the limitation to be used in the top level of the component. Are there any benefits of using `useDispatch` instead of `dispatch`?

I updated it for this file here: fc4eefdfd38086a99888d592b9d0bfe23004aec9

Reply:
I tried to search for a clear documented recommendation from WP, but I didn't find it. Basically, it'll have the same effect, but as we're accessing it through the components, a good practice, would be access these functions (`select`, `dispatch`) from features that expose it to be used in the components, like the `withDispatch`, `withSelect`, `useSelect` and `useDispatch`. If we check the Gutenberg code, we can notice that they're using the `dispatch` from `@wordpress/data` only in cases that are not components.

I'm really not sure why we have the dispatch exposed to be imported directly in the Gutenberg, probably something was easier for the `controls`, or something like that was implemented over Redux. But if we get the Redux as an example, there is no way to access a `dispatch` method directly.

Reply:
For `select` especially it is different than `useSelect` as the first one will simply fetch the data while the second one will also trigger a rerender of the component if the value of `useSelect` changes. I might be wrong but I think that these two should have distinct use cases.

Now the `dispatch` case seems weird as I can't see how `useDispatch` could cause a rerender (unless the callback changes?) so there seems to be no difference between them. To add a bit more to the confusion there is a third version of `dispatch` exposed by `@wordpress/data-controls` which is supposed to be used within controls according to [this article](https://unfoldingneurons.com/2020/wordpress-data-store-properties-controls) although `dispatch` from `@wordpress/data` can be used too.

> But if we get the Redux as an example, there is no way to access a dispatch method directly.

Isn't `store.dispatch()` part of the [public API](https://redux.js.org/basics/store#dispatching-actions) of Redux though? 

In any case I'll update it for the other file too although I am still baffled if there is an actual benefit other than being a hook.

Reply:
> For select especially it is different than useSelect as the first one will simply fetch the data while the second one will also trigger a rerender of the component if the value of useSelect changes. I might be wrong but I think that these two should have distinct use cases.

You're right! It memoizes the information when the dependencies are the same. So there is this extra advantage to use the `useSelect` hook instead of the `select` directly.

> Now the dispatch case seems weird as I can't see how useDispatch could cause a rerender (unless the callback changes?) so there seems to be no difference between them.

Yep! In the practice, I think we wouldn't have difference.

> To add a bit more to the confusion there is a third version of dispatch exposed by @wordpress/data-controls which is supposed to be used within controls according to this article although dispatch from @wordpress/data can be used too.

haha. Yep! I really don't know if there is some difference between these both `dispatch`. And I think the same happens to other functions like `apiFetch`, `select`...

> Isn't store.dispatch() part of the public API of Redux though?

The store will be a reference that you get when you create your store. So you wouldn't have something like `import { dispatch } from 'redux'`.

> In any case I'll update it for the other file too although I am still baffled if there is an actual benefit other than being a hook.

Maybe we could ask in the slack or in a GitHub issue to get an official Gutenberg recommendation about it. And maybe understand why we have the `dispatch` and `select` exposed directly. :)